src/Components/Layout.tsx

```diff
@@ -25,49 +26,107 @@ import { MenuView } from "./Menu.tsx";
+  timeFormat?: string;
 }
 
+var timeFormat: string = "en-US";
+
 export function Layout(props: Inputs) {
-  const [start, setStart] = useState(true);
-  const [recentlySelected, setRecentlySelected] = useState([]);
-  const [anchorEl, setAnchorEl] = useState(null);
+  // DATES
+  const [propertySelected, setPropertySelected] = useState(-1);
+  const [daysInMonth, setDaysInMonth] = useState([
+    new Date(new Date().getFullYear(), new Date().getMonth() + 1, 0).getDate(),
+    new Date(new Date().getFullYear(), new Date().getMonth() + 1, 0).getDate(),
+  ]);
+  const [dateRange, setDateRange] = useState(new DateRange());
+
+  // TAB LOGIC
+  const [tabSelected, setTabSelected] = useState(-1);
+  const [bodySubTabIndex, setBodySubTabIndex] = useState(0);
+
+  // DROPDOWN DATA
   const [termAnchorEl, setTermAnchorEl] = useState(null);
+  const [anchorEl, setAnchorEl] = useState(null);
   const [intervalAnchorEl, setIntervalAnchorEl] = useState(null);
-  const [quickSelectText, setQuickSelectText] = useState([
-    "last",
+  const [quickSelectContent, setQuickSelectContent] = useState([
+    "Last",
     "15 minutes",
   ]);
-  const [refreshIntervalUnits, setRefreshIntervalUnits] = useState("Minutes");
-  const [tabSelected, setTabSelected] = useState(-1);
-  const [propertySelected, setPropertySelected] = useState(-1);
+  const [relativeSelectContent, setRelativeSelectContent] = useState([
+    "15 minutes",
+    "ago",
+  ]);
+
+  // QUICK SELECT
+  const [recentlySelected, setRecentlySelected] = useState([[]]);
+
+  // TEXT FIELD INPUT
+  const [dateTextContents, setDateTextContents] = useState([
+    DateRange.formatAbsoluteDate(new Date()),
+    DateRange.formatAbsoluteDate(new Date()),
+  ]);
+  const [timeTextContents, setTimeTextContents] = useState([
+    new Date().toLocaleTimeString(timeFormat),
+    new Date().toLocaleTimeString(timeFormat),
+  ]);
 
+  // STYLING
   const [menuClass, setMenuClass] = useState("menu-closed");
   const [boxClass, setBoxClass] = useState("box-closed");
+
+  // TIMER
+  const [refreshIntervalUnits, setRefreshIntervalUnits] = useState("Minutes");
   const [refreshInterval, setRefreshInterval] = useState(-1);
   const [refreshIntervalEnabled, setRefreshIntervalEnabled] = useState(false);
-  const [dates, setDates] = useState([new Date(), new Date()]);
-  const [daysInMonth, setDaysInMonth] = useState([
-    new Date(new Date().getFullYear(), new Date().getMonth(), 0).getDate(),
-    new Date(new Date().getFullYear(), new Date().getMonth(), 0).getDate(),
-  ]);
   const [timerRunning, setTimerRunning] = useState(false);
+
+  // TEXT FIELD ERRORS
   const [menuError, setMenuError] = useState(false);
   const [dateError, setDateError] = useState([false, false]);
-  const [dateTextContents, setDateTextContents] = useState([
-    formatDateforDisplay(0),
-    formatDateforDisplay(1),
-  ]);
+  const [timeError, setTimeError] = useState([false, false]);
+
+  var timeIntervalText: string[][] = [

```

Comment:
Pull constants out of functions and into the top level so that they can be used as default parameters in props or overloaded as appropriate.